chain/network/src/peer_manager.rs

```diff
@@ -379,16 +400,62 @@ impl PeerManagerActor {
                 last_time_received_message: Instant::now(),
                 connection_established_time: Instant::now(),
                 peer_type,
+                protocol_version,
             },
         );
 
         self.process_edges(ctx, vec![new_edge.clone()]);
 
-        // TODO(MarX, #1363): Implement sync service. Right now all edges and known validators
-        //  are sent during handshake.
+        let wait_for_sync = 1;
+        checked_feature!(
+            "protocol_feature_routing_exchange_algorithm",
+            RoutingExchangeAlgorithm,
+            protocol_version,
```

Comment:
I am not sure whether we actually need a protocol change here. We can send different messages based on the protocol version of the peer (we receive that in handshake).

Reply:
This comment is still not addressed

Reply:
I don't understand your original comment. `protocol_version` variable is the protocol version of the peer you are connected to. So this code checks whenever peer version is high enough or not. If it is we do the new exchange and return.

I suspect there is some confusion on what `protocol_version` is. I will rename `protocol_version` to be `peer_protocol_version` to avoid confusion.

Reply:
Otherwise, we use old code path.

Reply:
@bowenwang1996

Reply:
My point was whether we need a protocol change here, but on second thought doing this makes the implementation cleaner and easier to reason about.